### PR TITLE
Clean up minor legacy UI in the unified dashboard widgets.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #4570

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { _x } from '@wordpress/i18n';
```
Comment (eugene-manuilov) [inline assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js:23]: This should be imported in the WordPress dependencies.

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
+/**
+ * Footer component for SearchFunnelWidget.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (eugene-manuilov) [inline assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js:4]: Also the current year is already 2022 😄. Please, update all your new files to use the correct year in the file headers.

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	const analyticsDeepLinks = {
+		users: analyticsVisitorsDeepLink,
+		goals: analyticsGoalsDeepLink,
+		'bounce-rate': analyticsVisitorsOverviewDeepLink,
+	};
+
+	const { service, id } = metrics?.[ selectedStats ];
+
+	return (
+		<Fragment>
```
Comment (eugene-manuilov) [inline assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js:137]: Maybe we don't need this `<Fragment>` element? Just return do it something like this:

```js
if ( service === 'search-console' ) {
    return (
        <SourceLink ... />
    );
}

return (
    <SourceLink href={ analyticsDeepLinks[ id ] } ... />
);
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @techanvil. Mostly looks good to me. Added a few comments. Once they are addressed, this PR should be good to go.

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js (hunk 1 above)
Comment (techanvil) [reply assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js]: Thanks, that's done :+1:

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js (hunk 2 above)
Comment (techanvil) [reply assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js]: Good spot, thanks! That's done :+1:

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js (hunk 3 above)
Comment (techanvil) [reply assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js]: Sounds good - that's done too. Cheers

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Pull request merged**